### PR TITLE
MCS: use cluster ID for ns lookup on exported service.

### LINKED ISSUE #12065: MultiClusterService:  Mon quorum fails on the peer cluster when mcs is enabled.
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
Mon quorum fails on the peer cluster when mcs is enabled.

**Expected behavior:**
Mon Quorum should not fail on the peer cluster

**How to reproduce it (minimal and precise):**

- Create two clusters with common CIDR.
- Connect clusters using Submariner globalnet.
- Enabled MulticlusterService on the first cluster.
- Enable MultiClusterService on the second cluster.
- Notice the mons on the second cluster.

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary

**Logs to submit**:

* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

  To get logs, use `kubectl -n <namespace> logs <pod name>`
  When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
  Read [GitHub documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Cluster Status to submit**:

* Output of krew commands, if necessary

  To get the health of the cluster, use `kubectl rook-ceph health`
  To get the status of the cluster, use `kubectl rook-ceph ceph status`
  For more details, see the [Rook Krew Plugin](https://rook.io/docs/rook/latest-release/Troubleshooting/krew-plugin)

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest-release/Troubleshooting/ceph-toolbox/#interactive-toolbox)):

### PR DESCRIPTION
**Pull request by sp98**
**Description of your changes:**
In MCS, to access the Service in a specific cluster, prefix the query with cluster-id so that we only query for the exported service on the local cluster. 

clusterid is a unique identity/name for a cluster and it should not overlap with any of the remote clusters in the clusterset as per the MCS API.

`ClusterID` should be provided by the user because there are multiple ways in which the clusterid can be configured as part of Submariner installation and assuming one could break the solution when deployed with a custom clusterid

**Which issue is resolved by this Pull Request:**
Resolves #12065

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by sp98** [comment]
testing:

kubectl get cephblockpools.ceph.rook.io mirrored-pool -n rook-ceph -o jsonpath='{.status.mirroringStatus.summary}'
{"daemon_health":"OK","health":"OK","image_health":"OK","states":{}}%

File: pkg/apis/ceph.rook.io/v1/types.go [hunk 1]
```diff
@@ -2207,6 +2207,10 @@ type MultiClusterServiceSpec struct {
 	// like Globalnet Submariner.
 	// +optional
 	Enabled bool `json:"enabled,omitempty"`
+
+	// ClusterID uniquely identifies a cluster. It is used as a prefix to nslookup exported
+	// services. For example: <clusterid>.<svc>.<ns>.svc.clusterset.local
```
Comment (travisn) [inline pkg/apis/ceph.rook.io/v1/types.go:2212]: How does the user know what to set the clusterID to? Would it be found in their kube context? And if the k8s dns lookup can work with it, isn't there a way rook can look up the cluster ID?

File: pkg/operator/ceph/cluster/mon/service.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ func (c *Cluster) createService(mon *monConfig) (*v1.Service, error) {
 
 func (c *Cluster) exportService(service *v1.Service, monDaemon string) (string, error) {
 	logger.Infof("exporting service %q", service.Name)
-	exportedIP, err := k8sutil.ExportService(c.ClusterInfo.Context, c.context, service)
+	exportedIP, err := k8sutil.ExportService(c.ClusterInfo.Context, c.context, service, c.spec.Network.MultiClusterService.ClusterID)
```
[the remark this answers is not in the document]
Comment (sp98) [reply pkg/operator/ceph/cluster/mon/service.go:78]: didn't get that question.

File: pkg/apis/ceph.rook.io/v1/types.go (hunk 1 above)
Comment (travisn) [reply pkg/apis/ceph.rook.io/v1/types.go:2212]: Ok, so the user must be aware of the clusterID and apply it the same both for submariner and in rook.

**Review by travisn — APPROVED** [review]

**Pull request merged**